src/roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -260,61 +260,53 @@ def __init__(self, file_path: PathType):
         self._io = NWBHDF5IO(str(file_path), mode="r")
         self.nwbfile = self._io.read()
 
+        assert "ophys" in self.nwbfile.processing, "Ophys processing module is not in nwbfile."
         ophys = self.nwbfile.processing.get("ophys")
-        if ophys is None:
-            raise Exception("could not find ophys processing module in nwbfile")
-        else:
-            # Extract roi_response:
-            fluorescence = None
-            dfof = None
-            any_roi_response_series_found = False
-            if "Fluorescence" in ophys.data_interfaces:
-                fluorescence = ophys.data_interfaces["Fluorescence"]
-            if "DfOverF" in ophys.data_interfaces:
-                dfof = ophys.data_interfaces["DfOverF"]
-            if fluorescence is None and dfof is None:
-                raise Exception("could not find Fluorescence/DfOverF module in nwbfile")
-            for trace_name in ["RoiResponseSeries", "Dff", "Neuropil", "Deconvolved"]:
-                trace_name_segext = "raw" if trace_name == "RoiResponseSeries" else trace_name.lower()
-                container = dfof if trace_name == "Dff" else fluorescence
-                if container is not None and trace_name in container.roi_response_series:
-                    any_roi_response_series_found = True
-                    setattr(
-                        self,
-                        f"_roi_response_{trace_name_segext}",
-                        DatasetView(container.roi_response_series[trace_name].data).lazy_transpose(),
-                    )
-                    if self._sampling_frequency is None:
-                        self._sampling_frequency = container.roi_response_series[trace_name].rate
-            if not any_roi_response_series_found:
-                raise Exception(
-                    "could not find any of 'RoiResponseSeries'/'Dff'/'Neuropil'/'Deconvolved'"
-                    "named RoiResponseSeries in nwbfile"
+
+        # Extract roi_responses:
+        fluorescence = None
+        df_over_f = None
+        any_roi_response_series_found = False
+        if "Fluorescence" in ophys.data_interfaces:
+            fluorescence = ophys.data_interfaces["Fluorescence"]
+        if "DfOverF" in ophys.data_interfaces:
+            df_over_f = ophys.data_interfaces["DfOverF"]
+        if fluorescence is None and df_over_f is None:
+            raise Exception("Could not find Fluorescence/DfOverF module in nwbfile.")
```

Comment:
Don't the `data_interfaces["Key"]` above return a KeyError when the "Fluoresence and DfOverF do not exist? Do them just return None?

Reply:
no, actually i didn't change that part, the intention was to declare above `df_over_f`, `fluorescence`=`None`, `None` and only override with the actual interface if we have that key. does that answer your question?

Reply:
Right, if you did not change then is not in the scope of this PR. Let's come back to this later. 

It is good that you eliminated the necessary outer loop for this PR (making the code flatter) is just that the diff then shows everything and it was hard to see what you did change. That line got my eye.